### DISCUSSION TITLE
1.0.0 - All systems go!

### DISCUSSION THREAD

**Discussion by aguilaair** [leoafarias/sidekick#218] [General]
<img src="https://user-images.githubusercontent.com/29983481/153480915-790f980a-553d-4124-9f65-70a0deb253fd.svg">

Hello everyone, 

Today we're excited to announce the release of Sidekick 1.0.0 🎉. It brings new features, new looks and huge optimizations, so let's get started.

# Refreshed Looks
<img src="https://user-images.githubusercontent.com/29983481/153480400-e3b31f2d-05e3-4164-900a-5094733ab754.svg" height="150" alt="Refreshed Looks Header">
Now, who doesn't love a refreshed, beautifully crafted UI and icon? To celebrate 1.0.0 we've concentrated on making the experience of using Sidekick even more tailored to the specific systems.

## macOS
In macOS we are excited to announce our new icon, which you can see in the header. It has been rebuilt to conform with macOS's other icons, and we guarantee it will feel right at home in your menu, with everything from it's subtle gradients to the shadows.

## Windows
<img src="https://user-images.githubusercontent.com/29983481/153481792-e4e0fb2c-8149-4363-900f-c8ce44070aa4.svg" height="150" alt="Sidekick Windows Mica Light ">
Windows has also received some love from us. In fact, Sidekick will now look somewhat different, with a beautiful Mica effect on Windows 11 and a curved main card which fits Microsoft's Fluent UI guidelines.

Let's also not forget about the fact that Sidekick is now available in the Microsoft Store, and has been for a couple of months. This means it is easier than ever to get started and set Sidekick up, simply by heading to the MS Store and clicking install. It has been great to see the reception of Sidekick in the Microsoft Store, with a 5-star rating and with over 200 downloads just in the first month from people all over the world.

Being added to the MS Store also brings some additional goodies to anyone who wants to keep using the MSIX to install Sidekick. The main advantage being that our installer is now signed, which means you no longer have to install our certificate in the root certificate authority database in order to install Sidekick. Simply open it and press install, it's that simple. Please do note that installing the signed certificate MSIX does remove the automatic update functionality as the MS Store handles it for us.

If you are using the MS Store 1.0.0 will be available shortly, but it may take a couple of days until the update rolls out to all devices. Hang tight!

# New Features

<img src="https://user-images.githubusercontent.com/29983481/153564654-a99a6773-5859-4bd4-8d20-f774ddc92804.svg" height="150" alt="Custom IDEs">
You can now select a custom application start your projects with! Simply head over to the IDE selection toggle and click on custom. Then, simply select the application you want to open your projects with.

Want to change the IDE again? Simply click on the custom button again to change your previous selection

<img src="https://user-images.githubusercontent.com/29983481/153583221-3b933856-7159-410c-9944-1b6a47802f57.svg" height="150" alt="Compatibility Checks">

We are taking the first step towards improving support for first time users of sidekick. As of 1.0.0 Sidekick will now check that all the required components are installed and correctly configured, as well as showing you the necessary commands to install everything with just one click.

# New Languages
This release brings some new translations thanks to all of you. Sidekick is now available in the following languages (in addition to any of the previously bundled ones): Japanese, Polish, Italian. Ukrainian(Слава Україні!), and Bengali

# Detailed changes

## Bug Fixes
* Fix crash on search bar and console widget by @federicoviceconti in https://github.com/leoafarias/sidekick/pull/159
* Fix CircularProgressIndicator never shown on first load by @federicoviceconti in https://github.com/leoafarias/sidekick/pull/161
* Performance improvements by @leoafarias in https://github.com/leoafarias/sidekick/pull/163
* Window improvements by @aguilaair in https://github.com/leoafarias/sidekick/pull/151
*  Fixed transition background update after theme change by @leoafarias in https://github.com/fluttertools/sidekick/pull/215

## New
* Add Japanese translation by @Chrg1001 in https://github.com/leoafarias/sidekick/pull/148
* Added polish translations by @LosDanieloss in https://github.com/leoafarias/sidekick/pull/152
* Adding italian language by @federicoviceconti in https://github.com/leoafarias/sidekick/pull/156
* Polish and improvements by @leoafarias in https://github.com/leoafarias/sidekick/pull/164
* feat fvm_global_dialog.dart add copy Flutter PATH value button by @Hu-Wentao in https://github.com/leoafarias/sidekick/pull/175
* Compatibility Check and Automatic Setup by @aguilaair in https://github.com/leoafarias/sidekick/pull/177
* Add Ukrainian language by @HighError in https://github.com/fluttertools/sidekick/pull/210
* Ukrainian language by @HighError in https://github.com/fluttertools/sidekick/pull/213
* Added Bengali language by @its-me-mahmud in https://github.com/fluttertools/sidekick/pull/208
* Feature/null safety migration by @leoafarias in https://github.com/fluttertools/sidekick/pull/188
* Feature/custom ide by @aguilaair in https://github.com/fluttertools/sidekick/pull/176

## Improvements
* Make playground text selectable by @aguilaair in https://github.com/leoafarias/sidekick/pull/162
* Remove pedantic and add flutter_lints by @aguilaair in https://github.com/leoafarias/sidekick/pull/169
* Improvements for windows by @aguilaair in https://github.com/leoafarias/sidekick/pull/168
* Upload new icons by @aguilaair in https://github.com/leoafarias/sidekick/pull/173
* Add artifacts to main build action by @aguilaair in https://github.com/leoafarias/sidekick/pull/181
* Updated mac os icon by @leoafarias in https://github.com/leoafarias/sidekick/pull/182
* Enable transparency on all Windows themes by @aguilaair in https://github.com/leoafarias/sidekick/pull/184
* Use actual Big Sur icon style by @aguilaair in https://github.com/leoafarias/sidekick/pull/185
* Null Safety Migration by @leoafarias in https://github.com/leoafarias/sidekick/pull/188
* Add artifacts to main build action by @aguilaair in https://github.com/fluttertools/sidekick/pull/181
* [ImgBot] Optimize images by @imgbot in https://github.com/fluttertools/sidekick/pull/180
* Enable transparency on all Windows themes by @aguilaair in https://github.com/fluttertools/sidekick/pull/184
* Improve window border on non-windows devices by @aguilaair in ade6666b357dca1e450ad39120f335801dd73f0e

## Docs
* Redesigned readme by @aguilaair in https://github.com/leoafarias/sidekick/pull/174
* Updated readme by @its-me-mahmud in https://github.com/fluttertools/sidekick/pull/209
* docs: add Chrg1001 as a contributor for translation by @allcontributors in https://github.com/leoafarias/sidekick/pull/149
* docs: add LosDanieloss as a contributor for translation by @allcontributors in https://github.com/leoafarias/sidekick/pull/153
* docs: add federicoviceconti as a contributor for translation by @allcontributors in https://github.com/leoafarias/sidekick/pull/157
* docs: add federicoviceconti as a contributor for bug by @allcontributors in https://github.com/leoafarias/sidekick/pull/160
* docs: add Hu-Wentao as a contributor for code by @allcontributors in https://github.com/leoafarias/sidekick/pull/179
* docs: add HighError as a contributor for translation by @allcontributors in https://github.com/fluttertools/sidekick/pull/211
* docs: add its-me-mahmud as a contributor for translation by @allcontributors in https://github.com/fluttertools/sidekick/pull/212

## Dependencies
* Add dependabot by @aguilaair in https://github.com/fluttertools/sidekick/pull/190
* Bump hive from 2.0.6 to 2.1.0 by @dependabot in https://github.com/fluttertools/sidekick/pull/192
* Bump build_runner from 2.1.7 to 2.1.8 by @dependabot in https://github.com/fluttertools/sidekick/pull/193
* Bump fvm from 2.2.6 to 2.3.0 by @dependabot in https://github.com/fluttertools/sidekick/pull/195
* Update dependency on constraints by @leoafarias in https://github.com/fluttertools/sidekick/pull/205
* Bump window_manager from 0.1.8 to 0.2.1 by @dependabot in https://github.com/fluttertools/sidekick/pull/200
* Bump path_provider from 2.0.1 to 2.0.9 by @dependabot in https://github.com/fluttertools/sidekick/pull/199
* Bump file_selector_windows from 0.0.2+1 to 0.8.2 by @dependabot in https://github.com/fluttertools/sidekick/pull/191
* Bump file_selector_macos from 0.0.4+1 to 0.8.2 by @dependabot in https://github.com/fluttertools/sidekick/pull/204
* Bump file_selector from 0.8.2+1 to 0.8.4+1 by @dependabot in https://github.com/fluttertools/sidekick/pull/206

## New Contributors
* @chrg1001 made their first contribution in https://github.com/fluttertools/sidekick/pull/148
* @LosDanieloss made their first contribution in https://github.com/fluttertools/sidekick/pull/152
* @federicoviceconti made their first contribution in https://github.com/fluttertools/sidekick/pull/156
* @dependabot made their first contribution in https://github.com/fluttertools/sidekick/pull/192
* @its-me-mahmud made their first contribution in https://github.com/fluttertools/sidekick/pull/208
* @HighError made their first contribution in https://github.com/fluttertools/sidekick/pull/210

**Full Changelog**: https://github.com/leoafarias/sidekick/compare/0.2.0...1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/fluttertools/sidekick/releases/tag/1.0.0'>1.0.0 - All systems go!</a>.</em>

**Comment by Hu-Wentao** [comment]
The picture looks really good. Is it made by PS or generated by code?

**Reply by aguilaair** [reply]
Thanks!! It's just made in Figma :)

**Reply by Hu-Wentao** [reply]
Oh, I thought you had an art friend. LOL
Is a powerful software~

**Reply by aguilaair** [reply]
https://www.figma.com/community/file/1106212737816275693
